datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonConfig.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hermes.datasetComparison
+
+import scopt.OptionParser
+
+/**
+  * This is a class for configuration provided by the command line parameters
+  *
+  * Note: scopt requires all fields to have default values.
+  *       Even if a field is mandatory it needs a default value.
+  */
+case class DatasetComparisonConfig(rawFormat: String = "xml",
+                                   rowTag: Option[String] = None,
+                                   csvDelimiter: String = ",",
+                                   csvHeader: Boolean = false,
+                                   newPath: String = "",
+                                   refPath: String = "",
+                                   outPath: String = "",
+                                   keys: Option[Seq[String]] = None) {
+  /**
+    * Checks if keys are defined
+    * @return True if the keys are defined
+    */
+  def hasKeysDefined: Boolean = keys.isDefined
+
+  /**
+    * Gets the keys
+    * @return Return the list of keys
+    */
+  def getKeys: Seq[String] = keys.get
+}
+
+object DatasetComparisonConfig {
+
+  def getCmdLineArguments(args: Array[String]): DatasetComparisonConfig = {
+    val parser = new CmdParser("spark-submit [spark options] TestUtils.jar")
+
+    parser.parse(args, DatasetComparisonConfig()) match {
+      case Some(config) => config
+      case _            => throw new IllegalArgumentException("Wrong options provided. List can be found above")
```

Comment:
```suggestion
      case _            => Failure(new IllegalArgumentException("Wrong options provided. List can be found above"))
```
I would make this function return a `Try` to make it obvious that this parsing can fail. It should be the client calling it, who's responsible for handling that case. What I always disliked about the `sys.exit` was that it kills your application like an assassin it's completely non-obvious from a user perspective that calling this funciton can result in death.

Reply:
I think I managed